modules/imgcodecs/src/loadsave.cpp

```diff
@@ -651,8 +651,14 @@ imdecode_( const Mat& buf, int flags, int hdrtype, Mat* mat=0 )
         if( !f )
             return 0;
         size_t bufSize = buf.cols*buf.rows*buf.elemSize();
-        fwrite( buf.ptr(), 1, bufSize, f );
-        fclose(f);
+        if( fwrite( buf.ptr(), 1, bufSize, f ) != bufSize )
+        {
+            CV_Error( CV_StsError, "failed to write image data to temporary file" );
```

Comment:
Good catch, thanks.  I see that this function both returns 0 to indicate an error in places, but elsewhere calls CV_Error, so I think this change at least doesn't make the situation worse.